File: Models/ViewModels/PostAmenityVM.cs
using GoWheels_WebAPI.Models.Entities;
using Microsoft.AspNetCore.Mvc.ModelBinding.Validation;

namespace GoWheels_WebAPI.Models.ViewModels
{
    public class PostAmenityVM
    {
        public int Id { get; set; }
        public int PostId { get; set; }
        public string? PostName { get; set; }
        public int AmenityId { get; set; }
        public string? AmenityName { get; set; }
        public bool IsDeleted { get; set; }
    }
}

File: FE/src/Model/PostAmenityVM.js
// src/models/PostAmenityVM.js

export default class PostAmenityVM {
    constructor(data = {}) {
      this.id = data.id || 0;
      this.postId = data.postId || 0;
      this.postName = data.postName || '';
      this.amenityId = data.amenityId || 0;
      this.amenityName = data.amenityName || '';
      this.amenityIconImage = data.amenityIconImage || '';
      this.isDeleted = data.isDeleted || false;
    }
  }
  